packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +27,115 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceDir,
+  cfnFile,
+  category,
+  filePath
+) {
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = await stateManager.getCustomPolicies(resourceDir);
+  if (customPolicies === undefined) return;
+
+  if(category === 'function') {
+    await addCustomPoliciesToCFNTemplateForFunction(customPolicies, cfnTemplate, filePath, {templateFormat} );
+  }
+  if(category === 'api') {
+    await addCustomPoliciesToCFNTemplateForContainer(customPolicies, cfnTemplate, filePath, {templateFormat});
+  }
+
+}
+
+export async function addCustomPoliciesToCFNTemplateForFunction(
```

Comment:
it seems like this method and the containers method below are nearly identical and can be combined into a single function. The policy and the resource name are the only thing that change between them, so I would suggest parameterizing these aspects and then only have the logic to validate / set the policies in one place

Reply:
> it seems like this method and the containers method below are nearly identical and can be combined into a single function. The policy and the resource name are the only thing that change between them, so I would suggest parameterizing these aspects and then only have the logic to validate / set the policies in one place

Done